packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/connector/ConnectionLine.tsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+import { ConnectionLineComponentProps, Position, getSmoothStepPath } from 'reactflow';
+import { useConnector } from './useConnector';
+
+export const ConnectionLine = ({ fromX, fromY, toX, toY }: ConnectionLineComponentProps) => {
+  const { connection } = useConnector();
+
+  const [state, setState] = useState({
+    frozenToX: 0,
+    frozenToY: 0,
+  });
+
+  useEffect(() => {
+    if (!!connection) {
+      setState((prevState) => ({ ...prevState, frozenToX: toX, frozenToY: toY }));
+    }
+  }, [connection]);
+
+  const [edgePath] = getSmoothStepPath({
+    sourceX: fromX,
+    sourceY: fromY,
+    sourcePosition: Position.Bottom,
+    targetX: state.frozenToX ? state.frozenToX : toX,
```

Comment:
`targetX: state.frozenToX ?? toX`

Reply:
I put || instead

Reply:
Do not use '||', it's confusing.
This only works because `state.frozenToX` and `state.frozenToY` are initialize to `0`.
Use the `iseFrozen` boolean instead.

Reply:
I put || because your solution with ?? was not working because like you said state.frozenToY is initialized to 0.
I'll revert back to the ternary operation then.

Reply:
Yes, I was wrong too at the beginning